packages/node/src/indexer/indexer.manager.ts

```diff
@@ -78,6 +78,7 @@ export class IndexerManager {
   async indexBlock(
     blockContent: BlockContent,
     runtimeVersion: RuntimeVersion,
+    filterDs?: (ds: SubqlProjectDs[]) => SubqlProjectDs[],
```

Comment:
Ideally we make no changes to the indexer.manager. We shouldn't be changing core functionality for tests

Reply:
Maybe rather than providing the projectService as a dependency we can provide the datasources here. 

It should be fairly easy to move the projectService.init elsewhere